File: backend/app/schemas/image.py
from datetime import datetime

from pydantic import BaseModel


class ImageResponse(BaseModel):
    id: int
    original_filename: str
    storage_key: str
    file_hash: str
    file_size: int
    width: int | None
    height: int | None
    mime_type: str
    folder_path: str
    dataset_id: int
    uploaded_by: int
    created_at: datetime

    model_config = {"from_attributes": True}


class ImageListResponse(BaseModel):
    images: list[ImageResponse]
    total: int


class FolderInfo(BaseModel):
    path: str
    name: str
    image_count: int
    subfolder_count: int


class FolderContentsResponse(BaseModel):
    current_path: str
    folders: list[FolderInfo]
    images: list[ImageResponse]
    total_images: int


class FolderUpdateRequest(BaseModel):
    old_path: str
    new_path: str


class FolderCreateRequest(BaseModel):
    path: str


class BatchDeleteRequest(BaseModel):
    image_ids: list[int]


class BatchMoveRequest(BaseModel):
    image_ids: list[int]
    target_folder: str


class BatchFolderDeleteRequest(BaseModel):
    paths: list[str]


class BatchFolderMoveRequest(BaseModel):
    paths: list[str]
    target_folder: str

File: frontend/src/types/image.ts
export interface ImageMeta {
  id: number;
  original_filename: string;
  file_size: number;
  width: number | null;
  height: number | null;
  mime_type: string;
  data_store_id: number;
  uploaded_by: number;
  created_at: string;
}

export interface FolderInfo {
  path: string;
  name: string;
  image_count: number;
  subfolder_count: number;
}

export interface FolderContentsResponse {
  current_path: string;
  folders: FolderInfo[];
  images: ImageMeta[];
  total_images: number;
}

export interface DataPoolItem {
  type: "folder" | "image" | "parent";
  key: string;
  folder?: FolderInfo;
  image?: ImageMeta;
}
